flink-runtime/src/test/java/org/apache/flink/runtime/state/TaskExecutorLocalStateStoresManagerTest.java

```diff
@@ -105,21 +96,11 @@ public void testCreationFromConfigDefault() throws Exception {
 
 		final Configuration config = new Configuration();
 
-		final ResourceID tmResourceID = ResourceID.generate();
-
-		TaskManagerServicesConfiguration taskManagerServicesConfiguration =
-			TaskManagerServicesConfiguration.fromConfiguration(config, MEM_SIZE_PARAM, InetAddress.getLocalHost(), true);
-
-		TaskManagerServices taskManagerServices = TaskManagerServices.fromConfiguration(
-			taskManagerServicesConfiguration,
-			tmResourceID,
-			Executors.directExecutor(),
-			MEM_SIZE_PARAM,
-			MEM_SIZE_PARAM);
+		TaskManagerServices taskManagerServices = createTaskManagerServices(config);
 
 		TaskExecutorLocalStateStoresManager taskStateManager = taskManagerServices.getTaskManagerStateStore();
 
-		String[] tmpDirPaths = taskManagerServicesConfiguration.getTmpDirPaths();
+		String[] tmpDirPaths = ConfigurationUtils.parseTempDirectories(config);
```

Comment:
Why is this change needed?

Reply:
In order to reuse the `TaskManagerServicesConfiguration` in `createTaskManagerServices`, otherwise  the creation of `TaskManagerServicesConfiguration` has to be outside of `createTaskManagerServices`. 

Comparing with reuse the creation of `TaskManagerServicesConfiguration`, I think it is cheaper to use `ConfigurationUtils.parseTempDirectories(config)` instead here.

Reply:
True, I have a little concern that we recreate it with a fixed. If the way of creation in `TaskManagerServicesConfiguration.fromConfiguration`, it can diverge with the test. Maybe break down `createTaskManagerServices(config)` into `createTaskManagerServicesConfiguration(config)`, `createTaskManagerServices(taskManagerServicesConfiguration)` and  in this case?

Reply:
Separating two methods might seem too thin for each method which only covers one sentence. 
Nevertheless I could accept both ways and separate it as you suggest. :)